### PR TITLE
provider/azurerm: Add disk_size_gb param to VM storage_os_disk

### PR DESCRIPTION
**Pull request by titanous**
TF_ACC=1 go test ./builtin/providers/azurerm -v -run=TestAccAzureRMVirtualMachine_basicLinuxMachine -timeout 120m
    === RUN   TestAccAzureRMVirtualMachine_basicLinuxMachine
    --- PASS: TestAccAzureRMVirtualMachine_basicLinuxMachine (540.83s)
    PASS
    ok  	github.com/hashicorp/terraform/builtin/providers/azurerm	540.841s

    TF_ACC=1 go test ./builtin/providers/azurerm -v -run=TestAccAzureRMVirtualMachine_withDataDisk -timeout 120m
    === RUN   TestAccAzureRMVirtualMachine_withDataDisk
    --- PASS: TestAccAzureRMVirtualMachine_withDataDisk (431.19s)
    PASS
    ok  	github.com/hashicorp/terraform/builtin/providers/azurerm	431.203s

### CODE REVIEW

**Comment by stack72** [comment]
Hi @titanous 

Thanks for the work here -- 2 small things:

1. Can you rebase from master - i have just merged the latest SDK changes
2. Will the disk_size_in_gb not have to be computed? If there is a disk_size in the api response from Azure and it's not classed as computed in our schema, then Terraform will try and change it

Paul

**Comment by titanous** [comment]
@stack72 

> Can you rebase from master - i have just merged the latest SDK changes

Done.

> Will the disk_size_in_gb not have to be computed? If there is a disk_size in the api response from Azure and it's not classed as computed in our schema, then Terraform will try and change it

The storage disk `disk_size_in_gb` is not marked as computed and no diff shows up when I run a subsequent `terraform plan`.

File: builtin/providers/azurerm/resource_arm_availability_set_test.go [hunk 1]
```diff
@@ -204,11 +206,11 @@ func testCheckAzureRMAvailabilitySetDestroy(s *terraform.State) error {
 var testAccAzureRMVAvailabilitySet_basic = `
 resource "azurerm_resource_group" "test" {
     name = "acctestRG-%d"
-    location = "West US"
+    location = "Canada Central"
```
Comment (carinadigital) [inline builtin/providers/azurerm/resource_arm_availability_set_test.go]: Did you mean to change the default test location?

Comment (titanous) [inline builtin/providers/azurerm/resource_arm_availability_set_test.go]: Yes, see the commit message here: 14169d8d91a3f4851a2a29c0ef308f787c5cda8c

Comment (titanous) [inline builtin/providers/azurerm/resource_arm_availability_set_test.go]: Actually, it looks like this commit is supposed to be on a separate PR. Fixing...

**Comment by carinadigital** [comment]
I can confirm @titanous findings about `disk_size_in_gb` on `storage_os_disk`. The other optional property `image_uri` also does not  produce a diff when you update it's value.

**Comment by stack72** [comment]
@carinadigital not sure what you mean here ?

**Comment by stack72** [comment]
Hi @titanous 

Thanks for the work here - this is working as you suggested :)

```
% make testacc TEST=./builtin/providers/azurerm TESTARGS='-run=TestAccAzureRMVirtualMachine_basicLinuxMachine'
==> Checking that code complies with gofmt requirements...
go generate $(go list ./... | grep -v /terraform/vendor/)
2016/10/25 17:11:46 Generated command/internal_plugin_list.go
TF_ACC=1 go test ./builtin/providers/azurerm -v -run=TestAccAzureRMVirtualMachine_basicLinuxMachine -timeout 120m
=== RUN   TestAccAzureRMVirtualMachine_basicLinuxMachine
--- PASS: TestAccAzureRMVirtualMachine_basicLinuxMachine (567.20s)
=== RUN   TestAccAzureRMVirtualMachine_basicLinuxMachine_disappears
--- PASS: TestAccAzureRMVirtualMachine_basicLinuxMachine_disappears (560.96s)
PASS
ok  	github.com/hashicorp/terraform/builtin/providers/azurerm	1128.173s
[stacko@Pauls-MacBook-Pro:~/Code/go/src/github.com/hashicorp/terraform on titanous-azurerm-add-os-disk-size]
% make testacc TEST=./builtin/providers/azurerm TESTARGS='-run=TestAccAzureRMVirtualMachine_withDataDisk'
==> Checking that code complies with gofmt requirements...
go generate $(go list ./... | grep -v /terraform/vendor/)
2016/10/25 17:34:28 Generated command/internal_plugin_list.go
TF_ACC=1 go test ./builtin/providers/azurerm -v -run=TestAccAzureRMVirtualMachine_withDataDisk -timeout 120m
=== RUN   TestAccAzureRMVirtualMachine_withDataDisk
--- PASS: TestAccAzureRMVirtualMachine_withDataDisk (683.58s)
PASS
ok  	github.com/hashicorp/terraform/builtin/providers/azurerm	683.596s
```
